src/Razor/Microsoft.AspNetCore.Razor.Language/src/DefaultRazorIntermediateNodeLoweringPhase.cs

```diff
@@ -1214,6 +1213,22 @@ public override void VisitMarkupElement(MarkupElementSyntax node)
                 base.VisitMarkupElement(node);
 
                 _builder.Pop();
+
+                bool LooksLikeAComponentName(string startTagName)
+                {
+                    if (char.IsUpper(startTagName, 0))
```

Comment:
would it be simpler if you can just do it as:

```C#
    UnicodeCategory category = char.GetUnicodeCategory(startTagName, 0);
    return category == UnicodeCategory.UppercaseLetter || UnicodeCategory.TitlecaseLetter || category == UnicodeCategory.OtherLetter;
```

Reply:
We're looking into putting `category == UnicodeCategory.TitlecaseLetter || category == UnicodeCategory.OtherLetter` behind a msbuild configurable property to prevent this from being a breaking change, so I believe we'll have to keep this separate.

Reply:
Are you going to add some more code here to handle the msbuild property value?

Reply:
Yep, working on adding the property now.

Would that be sufficient, we'd be missing the latin check from here. Is that acceptable?

https://github.com/dotnet/runtime/blob/bfaa457c804efb9247f495b7e04444b2bf216308/src/libraries/System.Private.CoreLib/src/System/Char.cs#L789-L793

Reply:
> Would that be sufficient, we'd be missing the latin check from here. Is that acceptable?

`char.GetUnicodeCategory` is already doing the Latin check too. The behavior would be exact as before.

https://github.com/dotnet/runtime/blob/7e67f8763d998ce0a626e91c647122cdc19d7c5c/src/libraries/System.Private.CoreLib/src/System/Char.cs#L835